include/libvfio-user.h

```diff
@@ -316,24 +313,28 @@ vfu_setup_device_reset_cb(vfu_ctx_t *vfu_ctx, vfu_reset_cb_t *reset);
  * Function that is called when the guest maps a DMA region. Optional.
  *
  * @vfu_ctx: the libvfio-user context
- * @iova: iova address
- * @len: length
+ * @iova: IOVA address and length
+ * @vaddr: virtual address and length if the region is mappable, which can be
+ *  different from IOVA due to alignment. If the region is not mappable then
+ *  @vaddr is set to NULL.
  * @prot: memory protection used to map region as defined in <sys/mman.h>
  */
-typedef void (vfu_map_dma_cb_t)(vfu_ctx_t *vfu_ctx,
-                                uint64_t iova, uint64_t len, uint32_t prot);
+typedef void (vfu_map_dma_cb_t)(vfu_ctx_t *vfu_ctx, const struct iovec *iova,
```

Comment:
if we're going to break compat, we might as well do the rename of these now too?

Reply:
Sure.